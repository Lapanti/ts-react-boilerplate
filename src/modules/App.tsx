import * as React from 'react';
import { Route, Switch, RouteComponentProps } from 'react-router-dom';
import { connect } from 'react-redux';
import styled from 'styled-components';

import routeEnter from '../components/HOC/routeEnter';
import HNClient from './hnClient/HNClient';
import { getStories } from './hnClient/HNClientReducer';
import PageNotFound from '../components/PageNotFound';
import StoryType from '../common/StoryType';

const MainContainer = styled.div`
  margin: 0 auto;
  max-width: 700px;
`;

export type AppProps = RouteComponentProps<undefined>;

export const App: React.StatelessComponent<AppProps> = () => (
  <MainContainer>
    <Switch>
      <Route
        path="/"
        exact
        component={routeEnter(HNClient, getStories(StoryType.NEW))}
      />
      <Route component={PageNotFound} />
    </Switch>
  </MainContainer>
);

export default connect<{}, undefined, AppProps>(() => ({}))(App);
