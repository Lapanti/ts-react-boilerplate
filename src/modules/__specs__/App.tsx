import * as React from 'react';
import { shallow } from 'enzyme';
import createHistory from 'history/createBrowserHistory';
import { App } from '../App';

describe('App', () => {
    const index = (
        <App
            match={{ params: undefined, isExact: true, path: '', url: '' }}
            location={{ pathname: '', search: '', state: {}, hash: '', key: '' }}
            history={createHistory()}
        />
    );
    const notFound = (
        <App
            match={{ params: undefined, isExact: false, path: '', url: '' }}
            location={{ pathname: '', search: '', state: {}, hash: '', key: '' }}
            history={createHistory()}
        />
    );

    it('should render correctly', () => {
        expect(shallow(index)).toMatchSnapshot();
        expect(shallow(notFound)).toMatchSnapshot();
    });
});
