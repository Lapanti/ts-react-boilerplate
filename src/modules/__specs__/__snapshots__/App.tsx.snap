// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App should render correctly 1`] = `
<styled.div>
  <Switch>
    <Route
      component={[Function]}
      exact={true}
      path="/"
    />
    <Route
      component={[Function]}
    />
  </Switch>
</styled.div>
`;

exports[`App should render correctly 2`] = `
<styled.div>
  <Switch>
    <Route
      component={[Function]}
      exact={true}
      path="/"
    />
    <Route
      component={[Function]}
    />
  </Switch>
</styled.div>
`;
